sdcm/utils/aws_utils.py

```diff
@@ -263,13 +265,20 @@ def get_common_params(params: dict, regions: List, credentials: List, services:
                 )
 
 
-def get_ec2_network_configuration(regions: list[str], availability_zones: list[str], params: dict):
+def get_ec2_network_configuration(regions: list[str], availability_zones: list[str], params: dict,
+                                  auto_availability_zone: bool = False):
```

Comment:
why `auto_availability_zone` is needed to be added all the way down, if params is being passed to the functions

just do the params.get('auto_availability_zone') down in this function

I think there are more places  `get_common_params` or `get_ec2_network_configuration` get called

Reply:
Yes, there is one more place i didn't add it intentionally since it is in scope of k8s. No problem adding it there as well. @vponomaryov can you please advise - can there be any future need for this enhancement in k8s?

Reply:
it's a feature related to the AWS backend, k8s probably doesn't needed it now (same as longevity doesn't needed it)

This feature is currently only for artifacts tests, and there no real need of changing k8s code.

i.e. you can encapsulate the change only to `get_ec2_network_configuration`, no change those function signatures

Reply:
fixed